src/App/Fossa/ArchiveUploader.hs

```diff
@@ -64,7 +66,18 @@ compressAndUpload apiOpts arcDir tmpDir dependency = do
 -- Using this information, it uploads each vendored dependency and queues a build for the dependency.
 archiveUploadSourceUnit :: (Has Diag.Diagnostics sig m, Has (Lift IO) sig m) => Path Abs Dir -> ApiOpts -> [VendoredDependency] -> m [Locator]
 archiveUploadSourceUnit baseDir apiOpts vendoredDeps = do
-  archives <- withSystemTempDir "fossa-temp" (uploadArchives apiOpts vendoredDeps baseDir)
+  -- Users with many instances of vendored dependencies may accidentally have complete duplicates. Remove them.
+  let uniqDeps = nub vendoredDeps
+
+  -- However, users may also have vendored dependencies that have duplicate names but are not complete duplicates.
+  -- These aren't valid and can't be automatically handled, so fail the scan with them.
+  let duplicates = duplicateNames uniqDeps
+  if not $ null duplicates
```

Comment:
[optional] there's a function in `Control.Monad` called `when :: Applicative f => Bool -> f () -> f ()` that does essentially this:

```hs
when :: Applicative f => Bool -> f () -> f ()
when b then' =
  if b
    then then'
    else pure ()
```
or pattern-match style:
```hs
when :: Applicative f => Bool -> f () -> f ()
when True then' = then'
when False _ = pure ()
```


which we can optionally use here:

```hs
when (not $ null duplicates) $
  Diag.fatalText (duplicateFailureBundle duplicates)
```